staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -315,9 +319,32 @@ func cleanVerb(verb string, request *http.Request) string {
 	if verb == "WATCHLIST" {
 		reportedVerb = "WATCH"
 	}
+	if verb == "PATCH" && request.Header.Get("Content-Type") == string(types.ApplyPatchType) {
+		reportedVerb = "APPLY"
+	}
 	return reportedVerb
 }
 
+func cleanDryRun(dryRun []string) string {
+	if err := validation.ValidateDryRun(nil, dryRun); err != nil {
+		return "invalid"
+	}
+
+	// Since dryRun could be valid with any arbitrarily long length
+	// we have to dedup and sort the elements before joining them together
+	dryRunSet := map[string]bool{}
+	for _, element := range dryRun {
+		dryRunSet[element] = true
```

Comment:
please also validate each element?

Reply:
the validate function is called above, they should have been validated already

Reply:
ah sgtm

Reply:
The elements are what's validated by validation.ValidateDryRun